### PR TITLE
modules: Add ModuleQuery

### PR DESCRIPTION
**Pull request by pkratoch**

### CODE REVIEW

File: include/libdnf/module/module_item.hpp [hunk 1]
```diff
@@ -52,7 +52,9 @@ struct ModuleItemId {
 // @replaces libdnf:module/ModuleItem.hpp:class:ModuleItem
 class ModuleItem {
 public:
-    bool operator==(const ModuleItem & r) const;
+    bool operator==(const ModuleItem & rhs) const noexcept { return id.id == rhs.id.id; }
```
Comment (j-mracek) [inline include/libdnf/module/module_item.hpp:55]: I have a question - is id mandatory for ModuleItem costructor?

File: include/libdnf/module/module_item.hpp [hunk 2]
```diff
@@ -165,17 +167,18 @@ class ModuleItem {
     // TODO(jmracek) Read a real decision from the modular solver
     bool is_active() const;
 
-private:
-    friend class ModuleSack;
-    friend class ModuleMetadata;
-    friend ::ModuleTest;
-
     ModuleItem(_ModulemdModuleStream * md_stream, const ModuleSackWeakPtr & module_sack, const std::string & repo_id);
```
Comment (kontura) [inline include/libdnf/module/module_item.hpp:170]: Should we keep this constructor private?
I believe it is not needed for `ModuleQuery` and I think API users don't have a way to make valid `_ModulemdModuleStream` anyway.

Comment (pkratoch) [reply include/libdnf/module/module_item.hpp:170]: It's not very well visible, because of how the changes are shown, but the constructor is not private.

Comment (kontura) [reply include/libdnf/module/module_item.hpp:170]: Yes, now it is not private, but I think it should be.

Comment (pkratoch) [reply include/libdnf/module/module_item.hpp:170]: Oh, sorry, I misunderstood, you are right

**Comment by j-mracek** [comment]
LGTM

**Pull request merged**